##### Exercise 3

###### a) Using the January 1976 - December 1979 four-year time period and the CAPM, generate variables measuring the GOLD-specific and the market risk premiums,and then estimate $\beta$ for GOLD. 

In [3]:
import numpy as np
import pandas as pd
from scipy import stats
import statsmodels.api as sm
from statsmodels.formula.api import ols

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


In [4]:
frame = pd.read_excel('chapter2data.xlsx', header=0, index_col='DATE')

In [5]:
gold   = frame.GOLD.loc  [pd.Timestamp('1976-01-01'):pd.Timestamp('1979-12-01')]
rkfree = frame.RKFREE.loc[pd.Timestamp('1976-01-01'):pd.Timestamp('1979-12-01')]
market = frame.MARKET.loc[pd.Timestamp('1976-01-01'):pd.Timestamp('1979-12-01')]

In [6]:
gold_specific       = gold - rkfree
market_risk_premium = market - rkfree

In [7]:
gold_slope, gold_intercept, gold_r_value, gold_p_value, gold_std_err = stats.linregress(gold_specific,
                                                                                        market_risk_premium)

In [8]:
print('alpha for GOLD:', gold_intercept)
print('betta for GOLD:', gold_slope)

alpha for GOLD: 0.0242045534692234
betta for GOLD: -0.23380356107998435


$\beta$ for GOLD is -0.2338

Compute 95% confidence interval for $\beta$

In [9]:
df_gold = pd.DataFrame({'x':market_risk_premium , 'y': gold_specific})
model_gold = ols("y ~ x", df_gold).fit()
model_gold.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.055
Model:                            OLS   Adj. R-squared:                  0.034
Method:                 Least Squares   F-statistic:                     2.672
Date:                Fri, 12 Oct 2018   Prob (F-statistic):              0.109
Time:                        04:12:54   Log-Likelihood:                 69.399
No. Observations:                  48   AIC:                            -134.8
Df Residuals:                      46   BIC:                            -131.1
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0259      0.009      2.927      0.005       0.008       0.044
x             -0.2348      0.144     -1.635      0.109      -0.524       0.054
==============================================================================
Omnibus:                        4.217   Durbin-Watson:                   1.450
Prob(Omnibus):                  0.121   Jarque-Bera (JB):                3.071
Skew:                           0.505   Prob(JB):                        0.215
Kurtosis:                       3.718   Cond. No.                         17.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Confidence interval for $\beta$ is [-0.524,0.054] 

Why might such an asset be particularly desirable to an investor who is attempting to reduce risk through diversification? What does this imply concerning the expected return on such an asset?

Gold is an asset that can be quickly sold in any situation.

###### b) Estimate the $\beta$ for GOLD using data from January 1980-December 1985 and construct a 95% confidence interval for $\beta$.

In [10]:
gold   = frame.GOLD.loc[pd.Timestamp('1980-01-01'):pd.Timestamp('1985-12-01')]
rkfree = frame.RKFREE.loc[pd.Timestamp('1980-01-01'):pd.Timestamp('1985-12-01')]
market = frame.MARKET.loc[pd.Timestamp('1980-01-01'):pd.Timestamp('1985-12-01')]

In [11]:
gold_specific = gold - rkfree
market_risk_premium = market - rkfree

In [12]:
gold_slope, gold_intercept, gold_r_value, gold_p_value, gold_std_err = stats.linregress(gold_specific,
                                                                                        market_risk_premium)
print('alpha for GOLD:', gold_intercept)
print('betta for GOLD:', gold_slope)

alpha for GOLD: 0.005724775562333638
betta for GOLD: 0.24836205036317202


$\beta$ for GOLD is 0.2484

In [13]:
df_gold = pd.DataFrame({'x':market_risk_premium , 'y': gold_specific})
model_gold = ols("y ~ x", df_gold).fit()
model_gold.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.101
Model:                            OLS   Adj. R-squared:                  0.088
Method:                 Least Squares   F-statistic:                     7.846
Date:                Fri, 12 Oct 2018   Prob (F-statistic):            0.00658
Time:                        04:12:57   Log-Likelihood:                 79.725
No. Observations:                  72   AIC:                            -155.4
Df Residuals:                      70   BIC:                            -150.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.0110      0.010     -1.152      0.253      -0.030       0.008
x              0.4058      0.145      2.801      0.007       0.117       0.695
==============================================================================
Omnibus:                       59.338   Durbin-Watson:                   1.683
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              369.487
Skew:                           2.362   Prob(JB):                     5.85e-81
Kurtosis:                      13.042   Cond. No.                         15.2
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Confidence interval for $\beta$ is [0.117,0.695] 

Comment on supply and demand shift factors possibly altering the $\beta$ of GOLD

After a little analysis we could notice an increase in price of gold as an asset, which lasted for at least 1o years from 1970 to 1980. 

This situation arose because the price was constantly manipulated - suppressed - during the 1960s. When the manipulations ceased, prices rose strongly and quickly in order to reach the level to which they themselves would rise without suppression.

After a period of moderate inflation, which began in 1980, the trend in the gold market altered. 

##### Exercise 4

###### a) We need to show that using "the incorrect" reciprocal regression equation, one can still find the original CAPM parameters

In [14]:
Delta  = frame.DELTA.loc [pd.Timestamp('1983-01-01'):pd.Timestamp('1987-12-01')]
rkfree = frame.RKFREE.loc[pd.Timestamp('1983-01-01'):pd.Timestamp('1987-12-01')]
market = frame.MARKET.loc[pd.Timestamp('1983-01-01'):pd.Timestamp('1987-12-01')]

In [15]:
delta_risk_premium = Delta - rkfree
market_risk_premium = market - rkfree

In [16]:
delta_wrong_slope, delta_wrong_intercept, delta_wrong_r_value, delta_wrong_p_value, delta_wrong_std_err = stats.linregress( market_risk_premium,
                                                                                                          delta_risk_premium)

In [17]:
print('inversed delta for DELTA:', delta_wrong_intercept)
print('inversed gamma for DELTA:', delta_wrong_slope)
print('checking the hypothesis, that gamma is 0, p-value is', delta_wrong_p_value)

inversed delta for DELTA: -0.004296310359829957
inversed gamma for DELTA: 0.6395249978002865
checking the hypothesis, that gamma is 0, p-value is 0.001313131332783861


We reject the hypothesis that gamma equals to 0

In [18]:
delta = -delta_wrong_intercept/delta_wrong_slope
print(delta)

0.006717970954392039


In [19]:
gamma = 1/delta_wrong_slope
print(gamma)

1.5636605346774641


In [20]:
delta_slope, delta_intercept, delta_r_value, delta_p_value, delta_std_err = stats.linregress(delta_risk_premium,
                                                                                             market_risk_premium)

In [21]:
print('alpha for DELTA:', delta_intercept)
print('betta for DELTA:', delta_slope)
print('checking the hypothesis, that beta is 0, p-value is', delta_p_value)

alpha for DELTA: 0.003701213838445564
betta for DELTA: 0.25695083963856374
checking the hypothesis, that beta is 0, p-value is 0.001313131332783861


We reject the hypothesis, that beta equals to 0 (p-value is less that 0.05)

We can see, that coeffs, which were calculated using the wrong method, can not be changed to the right ones by any manipulations.

In [313]:
df_delta = pd.DataFrame({'x':market_risk_premium , 'y':delta_risk_premium})
model_delta = ols("y ~ x", df_delta).fit()
model_delta.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.164
Model:                            OLS   Adj. R-squared:                  0.150
Method:                 Least Squares   F-statistic:                     11.41
Date:                Fri, 12 Oct 2018   Prob (F-statistic):            0.00131
Time:                        04:01:51   Log-Likelihood:                 62.531
No. Observations:                  60   AIC:                            -121.1
Df Residuals:                      58   BIC:                            -116.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.0043      0.011     -0.383      0.703      -0.027       0.018
x              0.6395      0.189      3.377      0.001       0.260       1.019
==============================================================================
Omnibus:                        0.111   Durbin-Watson:                   2.160
Prob(Omnibus):                  0.946   Jarque-Bera (JB):                0.003
Skew:                          -0.004   Prob(JB):                        0.999
Kurtosis:                       2.967   Cond. No.                         16.9
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

$R^2$ equals to 0.164

In [314]:
df_delta = pd.DataFrame({'y':market_risk_premium , 'x':delta_risk_premium})
model_deltAa = ols("y ~ x", df_delta).fit()
model_delta.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.164
Model:                            OLS   Adj. R-squared:                  0.150
Method:                 Least Squares   F-statistic:                     11.41
Date:                Fri, 12 Oct 2018   Prob (F-statistic):            0.00131
Time:                        04:01:51   Log-Likelihood:                 62.531
No. Observations:                  60   AIC:                            -121.1
Df Residuals:                      58   BIC:                            -116.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.0043      0.011     -0.383      0.703      -0.027       0.018
x              0.6395      0.189      3.377      0.001       0.260       1.019
==============================================================================
Omnibus:                        0.111   Durbin-Watson:                   2.160
Prob(Omnibus):                  0.946   Jarque-Bera (JB):                0.003
Skew:                          -0.004   Prob(JB):                        0.999
Kurtosis:                       2.967   Cond. No.                         16.9
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

$R^2$ also equals ro 0.164

But, as we see, $R^2$ happens to be the same

**_d) Why $R_1^2$ from the correct equation is identical to the $R_2^2$ from the incorrect equation? Show that $R^2 = b_y * g = b_y / b_x$_**

1) $$R_1^2 = 1 - \frac{V(y|x)}{V(y)} = 1 - \frac{V(x|y)}{V(x)} = R_2^2$$

2) From b by construction: $b_x = 1 / g$ => $b_y * g = b_y / b_x$

$$b_y = \frac{\sum x_i*y_i}{\sum x_i^2}$$
$$g = \frac{\sum x_i*y_i}{\sum y_i^2}$$
$$b_y * g = \frac{\sum x_i*y_i}{\sum x_i^2} * \frac{\sum x_i*y_i}{\sum y_i^2} = \frac{(\sum x_i*y_i)^2}{\sum x_i^2 * \sum y_i^2} = R^2$$

**Verify numerically that this relationship among $R^2$, b_y, g and b_x occurs with your data for Delta Airlines**

In [22]:
r2 = delta_wrong_r_value ** 2
g = delta_wrong_slope
b_x = 1 / g
b_y = r2 * b_x
print('b_y get from the relationship:', b_y, '=', r2, '*', b_x)
print('actual b_y:', delta_slope)

b_y get from the relationship: 0.25695083963856374 = 0.16432648515463424 * 1.5636605346774641
actual b_y: 0.25695083963856374


**e) Which estimate of $\beta$ do you prefer - the smaller $b_y$ or the larger $b_x$?**

Well, we defined $b_y$ as an estimator of $\beta$...do we have a choice? Larger b doesn't mean better b as we have already seen in exercise 2, because larger b doesn't mean larger $R^2$.